vshard/router/init.lua

```diff
@@ -1709,6 +1695,20 @@ local function router_api_call_unsafe(func, router, ...)
     return func(router, ...)
 end
 
+local function router_cfg_fiber_safe(router, cfg, is_reload)
+    if router.is_cfg_in_progress then
+        error(lerror.vshard(lerror.code.ROUTER_CFG_IS_IN_PROGRESS, router.name))
+    end
+
+    router.api_call_cache = router_api_call_unsafe
```

Comment:
Wouldn't it work to set the api call cache in the same place where you set `is_enabled` on reload now?

Reply:
No problem, we can place it there